api/app/Http/Controllers/DashboardController.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Announcement;
+use App\Models\PermissionRole;
+use App\Models\User;
+use Illuminate\Http\Request;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\Auth;
+
+class DashboardController extends Controller
+{
+    public function index() 
+    {
+        $announcements = Announcement::all();
+
+        return response()->json($announcements);
+    }
+
+    public function store(Request $request)
+    {
+        $user = Auth::user();
+        
+        $user_fullname = $user['fullname'];
+        $role_users = $user->roleUsers;
```

Comment:
Since you only need the role_id I think mas direct if we use first() to get it out from the array format
```$role_id = $user->roleUsers->first()->role_id```